root@8b6c9cbb837f:/workspace# cd DL_healthcare
bash: cd: DL_healthcare: No such file or directory
root@8b6c9cbb837f:/workspace# cd DL-healthcare
root@8b6c9cbb837f:/workspace/DL-healthcare# pip install seaborn
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Collecting seaborn
  Downloading seaborn-0.13.2-py3-none-any.whl.metadata (5.4 kB)
Requirement already satisfied: numpy!=1.24.0,>=1.20 in /usr/local/lib/python3.10/dist-packages (from seaborn) (1.24.4)
Requirement already satisfied: pandas>=1.2 in /usr/local/lib/python3.10/dist-packages (from seaborn) (2.2.2)
Requirement already satisfied: matplotlib!=3.6.1,>=3.4 in /usr/local/lib/python3.10/dist-packages (from seaborn) (3.9.2)
Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (1.3.0)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (4.54.1)
Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (1.4.7)
Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (23.2)
Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (10.4.0)
Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (3.1.4)
Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.10/dist-packages (from matplotlib!=3.6.1,>=3.4->seaborn) (2.9.0.post0)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=1.2->seaborn) (2023.4)
Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas>=1.2->seaborn) (2024.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.7->matplotlib!=3.6.1,>=3.4->seaborn) (1.16.0)
Downloading seaborn-0.13.2-py3-none-any.whl (294 kB)
Installing collected packages: seaborn
Successfully installed seaborn-0.13.2
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.

[notice] A new release of pip is available: 24.2 -> 25.1.1
[notice] To update, run: python -m pip install --upgrade pip
root@8b6c9cbb837f:/workspace/DL-healthcare# python src/main.py --class_config 2
🚀 Using device: cuda
📱 Starting model training...
📊 Training for 30 epochs with 27972 batches per epoch.
Epoch 1/30: 100%|███████████████████████████████████████████████████████████████████████████████| 27972/27972 [07:32<00:00, 61.84it/s, loss=0.0060, accuracy=0.9970]
💚 Validation Loss after Epoch 1: 0.0059
Epoch 2/30: 100%|███████████████████████████████████████████████████████████████████████████████| 27972/27972 [07:17<00:00, 63.96it/s, loss=0.0008, accuracy=0.9974]
💚 Validation Loss after Epoch 2: 0.0057
Epoch 3/30: 100%|███████████████████████████████████████████████████████████████████████████████| 27972/27972 [07:00<00:00, 66.52it/s, loss=0.0035, accuracy=0.9974]
💚 Validation Loss after Epoch 3: 0.0058
Epoch 4/30: 100%|███████████████████████████████████████████████████████████████████████████████| 27972/27972 [07:02<00:00, 66.23it/s, loss=0.0001, accuracy=0.9975]
💚 Validation Loss after Epoch 4: 0.0053
Epoch 18/30: 100%|██████████████████████████████████████████████████████████████████████████████| 27972/27972 [07:14<00:00, 64.31it/s, loss=0.0038, accuracy=0.9977]
💚 Validation Loss after Epoch 18: 0.0050
⏹ Early stopping triggered.
/workspace/DL-healthcare/src/main.py:74: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(os.path.join(output_dir, "best_model.pth")))

🔍 Final Metrics on Test Set:
Accuracy: 0.9977
Precision: 0.9977
Recall: 0.9977
F1 Score: 0.9977

📊 Classification Report:
               precision    recall  f1-score   support

      Benign       0.96      0.96      0.96     80835
      attack       1.00      1.00      1.00   2944916

    accuracy                           1.00   3025751
   macro avg       0.98      0.98      0.98   3025751
weighted avg       1.00      1.00      1.00   3025751


🗙 Confusion Matrix:
 [[  77528    3307]
 [   3597 2941319]]
root@8b6c9cbb837f:/workspace/DL-healthcare# 